python/ray/tests/test_signal.py

```diff
@@ -279,3 +300,32 @@ def send_signals(self, value, count):
     ray.get(a.send_signals.remote(signal_value, count))
     result_list = receive_all_signals([a], timeout=5)
     assert len(result_list) == count
+
+
+
+def test_signal_on_node_failure(two_node_cluster):
+    """Test actor checkpointing on a remote node."""
+
+    class ActorSignal(object):
+        def __init__(self):
+            pass
+
+        def local_plasma(self):
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+    # Place the actor on the remote node.
+    cluster, remote_node = two_node_cluster
+    actor_cls = ray.remote(max_reconstructions=0)(ActorSignal)
+    actor = actor_cls.remote()
+    # Try until we put an actor on a different node.
+    while (ray.get(actor.local_plasma.remote()) !=
+           remote_node.plasma_store_socket_name):
+        actor = actor_cls.remote()
```

Comment:
Instead of retrying, we can use custom resource to make sure the actor is deployed on the remote node.